modules/system/assets/js/framework.js

```diff
@@ -41,23 +41,36 @@ if (window.jQuery.request !== undefined) {
         $triggerEl.trigger(_event, context)
         if (_event.isDefaultPrevented()) return
 
-        var data = [$form.serialize()],
-            loading = options.loading !== undefined ? options.loading : null,
+        var loading = options.loading !== undefined ? options.loading : null,
             isRedirect = options.redirect !== undefined && options.redirect.length,
-            useFlash = options.flash !== undefined
+            useFlash = options.flash !== undefined,
+            sendFiles = $el.is(':file') || ($el[0] == $form[0] && $form.attr('enctype') == 'multipart/form-data') || options.sendFiles,
+            data = $triggerEl.find(':input').addBack(':input').serializeArray()
```

Comment:
This serializes the form to an array of objects with name and value properties. Rather than serializing the form directly, it serializes the child inputs of $triggerEl which is either the form or the actual element if not using a form. Doing it this way also allows the $triggerEl to be a div when not using a form, for example:
```html
<div id="login">
  <input type="text" name="username">
  <input type="password" name="password">
  <button type="button" onClick="$('#login').request('onLogin')">Login</button>
</div>
```
In this example, $triggerEl would be the div and the child inputs username and password would be submitted. This only works when the inputs are not part of a form.